client/controller/shared.js

```diff
@@ -37,31 +37,60 @@ export function makeLayoutMiddleware( LayoutComponent ) {
 	};
 }
 
+/***
+ * Replaces stylesheet element gracefully
+ *
+ * If we set the `href` attribute of `link` directly, the browser will remove
+ * the previous stylesheet and will start loading the new one, but in the meantime
+ * we won't have all of the sizes the old one had.
+ *
+ * To prevent that, we first load a new stylesheet and then remove the old one
+ *
+ * @param {DOMElement} linkElement Link to change
+ * @param {String} newHref new url to load
+ */
+function replaceStlyesheet( linkElement, newHref ) {
+	const newLink = Object.assign( document.createElement( 'link' ), {
+		rel: 'stylesheet',
+		type: 'text/css',
+		href: newHref,
+	} );
+
+	if ( linkElement.id ) {
+		newLink.id = linkElement.id;
+	}
+
+	function deleteOldLink() {
+		linkElement.parentElement.removeChild( linkElement );
+		newLink.removeEventListener( 'load', deleteOldLink );
+	}
+
+	newLink.addEventListener( 'load', deleteOldLink );
+
+	linkElement.parentElement.appendChild( newLink );
+}
+
 function loadCSS( cssUrl, elementId = null ) {
 	if ( typeof document === 'undefined' ) {
 		return;
 	}
 
-	let link;
 	if ( elementId ) {
-		link = document.getElementById( elementId );
-	}
+		const link = document.getElementById( elementId );
 
-	if ( link ) {
```

Comment:
This check is necessary because we might want to call `loadCSS( cssUrl, 'section-css' );` without `<link id="section-css" .../>` being in the page